### DISCUSSION TITLE
Is it possible to perfectly reproduce the correct Llama 2 Chat prompt structure in KoboldCPP?

### DISCUSSION THREAD

**Discussion by cgessai** [LostRuins/koboldcpp#332] [Q&A]
This is the prompt structure for Llama 2 Chat, [from here](https://www.reddit.com/r/LocalLLaMA/comments/155po2p/get_llama_2_prompt_format_right/):
```
<s>[INST] <<SYS>>
{{ system_prompt }}
<</SYS>>

{{ user_message }} [/INST]
``` 
Here's an example of what a correct prompt would look like:
```
<s>[INST] <<SYS>>
You are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe.  Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.

If a question does not make any sense, or is not factually coherent, explain why instead of answering something not correct. If you don't know the answer to a question, please don't share false information.
<</SYS>>

There's a llama in my garden 😱 What should I do? [/INST]
```

Is it possible to 100% reproduce that exact structure in KoboldCPP?  If so, how?

**Comment by LostRuins** [comment]
Not automatically, but you can certainly copy these tags into story mode and trigger generation from there. 

Otherwise, if you're using instruct mode, you can set the Instruct start tag to `[INST] ` and the end tag to `[/INST] ` with the spaces included, and it should work reasonably well. LLAMA2 is smart enough that even the default Alpaca/Vicuna style tags should also work.
